src/modules/auth/auth.resolver.ts

```diff
@@ -80,4 +83,40 @@ export class AuthResolver {
       message: 'registration failed',
     };
   }
+
+  @Mutation(() => Result, { description: 'Change password' })
+  @UseGuards(GqlAuthGuard)
+  async changePassword(
+    @Context() cxt: any,
+    @Args('oldPassword') oldPassword: string,
+    @Args('newPassword') newPassword: string,
+  ): Promise<Result> {
```

Comment:
let's add try-catch block in the resolver to catch any unexpected error, and return the error text in response